src/common/strtol.cc

```diff
@@ -28,6 +28,17 @@ long long strict_strtoll(const char *str, int base, std::string *err)
   errno = 0; /* To distinguish success/failure after call (see man page) */
   long long ret = strtoll(str, &endptr, base);
 
+  if (endptr == str
```

Comment:
@tchaikov 
Could be that the FreeSD version is not 100% compliant...
Reason for this change is that in a test the "wrong" error messages was returned.
An that was because `ret == 0` and `errno == EINVAL` Which triggered the first case.
Linux however chooses to not set `errno != 0` but has `str == endptr` and selects the second case.
FreeBSD does indeed also set:
``
`If there were no digits at all, however, strtoll()
     stores the original value of nptr in *endptr.
```
So perhaps just reversing the tests would already work.
I'll test that and adjust the PR if that works